frontend/src/api/api.ts

```diff
@@ -0,0 +1,203 @@
+// Still a few kinks to be ironed out - TS sees no issue but our linter does
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+// Response return is read as any by our linter for some reason
+/* eslint-disable @typescript-eslint/no-unsafe-return */
+import { AxiosResponse } from 'axios';
+import { SortDirection } from '@tanstack/react-table';
+import { CLValueParsers } from 'casper-js-sdk';
+import { formatDate, formatTimeAgo, loadConfig } from 'src/utils';
+import { createBaseApi } from './base-api';
+import { ApiData, DeployStatus } from './types';
+import { determineDeploySessionData, isValidPublicKey } from './utils';
+import { JsonDeploySession } from './missing-sdk-types';
+
+const { webServerUrl } = loadConfig();
+
+const createApi = (baseUrl: string) => {
+  const middlewareApi = createBaseApi(baseUrl);
+
+  return {
+    block: {
+      async getBlocks(tableParams: {
+        from?: number;
+        sort_by?: string;
+        order_by?: SortDirection;
+        count?: number;
+      }) {
+        type Response = AxiosResponse<ApiData.Blocks>;
+
+        const response = await middlewareApi.get<Response>('/blocks', {
+          params: { ...tableParams, count: tableParams.count ?? 10 },
```

Comment:
NIT: do you think we should slap that `10` in a constant variable at the top of this file?

Reply:
Ah, there's already a constant var for this so I just used that!